editor/components/block-switcher/index.js

```diff
@@ -25,41 +24,14 @@ import { getBlock } from '../../selectors';
 const { DOWN } = keycodes;
 
 function BlockSwitcher( { blocks, onTransform } ) {
-	if ( ! blocks || ! blocks[ 0 ] ) {
-		return null;
-	}
-	const isMultiBlock = blocks.length > 1;
-	const sourceBlockName = blocks[ 0 ].name;
+	const allowedBlocks = getPossibleBlockTransformations( blocks );
 
-	if ( isMultiBlock && ! every( blocks, ( block ) => ( block.name === sourceBlockName ) ) ) {
+	if ( ! allowedBlocks.length ) {
 		return null;
 	}
 
+	const sourceBlockName = blocks[ 0 ].name;
 	const blockType = getBlockType( sourceBlockName );
```

Comment:
Kinda makes me wish that `<BlockIcon />` could render via the name of the block, so we wouldn't have to get the full type ahead of type in usage.

Reply:
Yes, in most cases we have the block, so it makes sense to receive icon but it would make sense to also be possible to pass the name of the block. I will propose this change in a separate PR.